File: 78-subsets/subsets.java
class Solution {
    List<List<Integer>> result = new ArrayList<>();

    public List<List<Integer>> subsets(int[] nums) {
        backtrack(new ArrayList<>(), 0, nums);
        return result;
        
    }
    private void backtrack(List<Integer> path, int start, int[] nums) {
        result.add(new ArrayList<>(path)); 

        for (int i = start; i < nums.length; i++) {
            path.add(nums[i]);                    
            backtrack(path, i + 1, nums);         
            path.remove(path.size() - 1);        
        }
    }
}

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        output = []
        path = []

        def backtrack(i):
            if i == len(nums):
                output.append(path.copy())
                return
            
            path.append(nums[i]) #include
            backtrack(i+1)
            path.pop()

            backtrack(i+1)


        backtrack(0)
        return output

